ui/perfherder/graphs/TestDataModal.jsx

```diff
@@ -98,6 +98,53 @@ export default class TestDataModal extends React.Component {
     this.processOptions();
   }
 
+  addRelatedApplications = async params => {
+    const { relatedSeries: relatedSignature } = this.props.options;
+    const { errorMessages } = this.state;
+    let relatedTests = [];
+
+    const response = await PerfSeriesModel.getSeriesList(
+      relatedSignature.repository_name,
+      params,
+    );
+
+    if (!response.failureStatus) {
+      relatedTests = response.data.filter(signature => {
+        const [signName, signApplication, relSignName, relSignApplication] = [
```

Comment:
I think `[signName, signApplication, relSignName, relSignApplication]` can be removed and used the fields from `signature` and `relatedSignature` directly. The code would be straightforward and shorter. And this would be safe, since `excise` function does not modify the received parameters, but returns copies.